drivers/soundwire/intel.c

```diff
@@ -1019,7 +1021,8 @@ static int intel_prepare(struct snd_pcm_substream *substream,
 			goto err;
 	}
 
-	ret = sdw_prepare_stream(dma->stream);
+	if (cpu_dai == dai)
```

Comment:
nit-pick: maybe call it first_cpu_dai and add a comment to explain that this makes sure the stream state is consistent.

Reply:
or on line 978
struct snd_soc_dai *cpu_dai = rtd->cpu_dais[0];

Reply:
Applied both. Thanks @plbossart @ranj063